ql/src/java/org/apache/hadoop/hive/ql/exec/ColumnStatsUpdateTask.java

```diff
@@ -355,8 +360,32 @@ public int execute() {
     } catch (Exception e) {
       setException(e);
       LOG.info("Failed to persist stats in metastore", e);
+      int errorCode = ErrorMsg.getErrorMsg(e.getMessage()).getErrorCode();
+      try {
+        ReplicationMetricCollector metricCollector = work.getMetricCollector();
+        if (errorCode > 40000) {
+          //in case of replication related task, dumpDirectory should not be null
+          String dumpDirectory = work.getDumpDirectory();
+          if(dumpDirectory != null) {
+            Path nonRecoverableMarker = new Path(dumpDirectory, ReplAck.NON_RECOVERABLE_MARKER.toString());
+            org.apache.hadoop.hive.ql.parse.repl.dump.Utils.writeStackTrace(e, nonRecoverableMarker, conf);
+            if(metricCollector != null){
+              metricCollector.reportStageEnd(getName(), Status.FAILED_ADMIN, nonRecoverableMarker.toString());
+            }
+          }
+          if(metricCollector != null){
+            metricCollector.reportStageEnd(getName(), Status.FAILED_ADMIN, null);
+          }
+        } else {
+          if(metricCollector != null){
+            work.getMetricCollector().reportStageEnd(getName(), Status.FAILED);
+          }
+        }
+      } catch (SemanticException ex) {
```

Comment:
catch all exception so that metric collection failure doesn't lead to failure of task